### DISCUSSION TITLE
is realtime audio manipulation possible with this library?

### DISCUSSION THREAD

**Discussion by actuallytomato** [DBraun/DawDreamer#105] [Q&A]
like the title says, is it possible to manipulate audio and play it back in realtime? for example, generating a sine frequency and playing it through the speakers when a midi key event is recieved? 

i know how midi works in python, im just wondering how to get the audio output from the engine without having to render a certain length of audio. all the examples show rendering a section and then saving it to a wav file. 

thanks

**Comment by DBraun** [comment]
It's not a feature, but you can track its progress here in case if ever gets done https://github.com/DBraun/DawDreamer/issues/50

Meanwhile, depending on what you desire, you can currently get a good interactive (but non-realtime) workflow with Jupyter notebooks and `IPython.display.Audio`.
